x-pack/plugins/search_inference_endpoints/public/components/all_inference_endpoints/tabular_page.tsx

```diff
@@ -0,0 +1,167 @@
+    }),
+    [inferenceEndpoints, queryParams]
+  );
+
+  const tableOnChangeCallback = useCallback(
+    ({ page, sort }) => {
+      let newQueryParams = queryParams;
+      if (sort) {
+        newQueryParams = {
+          ...newQueryParams,
+          sortField: getSortField(sort.field),
+          sortOrder: sort.direction,
+        };
+      }
+      if (page) {
+        newQueryParams = {
+          ...newQueryParams,
+          page: page.index + 1,
+          perPage: page.size,
+        };
+      }
+      setQueryParams(newQueryParams);
+    },
+    [queryParams, setQueryParams]
+  );
+
+  const tableData: InferenceEndpointUI[] = useMemo(() => {
+    return inferenceEndpoints.map((endpoint) => ({
+      endpoint: endpoint.model_id,
+      provider: endpoint.service,
+      type: endpoint.task_type,
+    }));
+  }, [inferenceEndpoints]);
+
+  const sortedTableData: InferenceEndpointUI[] = useMemo(() => {
+    return [...tableData].sort((a, b) => {
+      const aValue = a[queryParams.sortField];
+      const bValue = b[queryParams.sortField];
+
+      if (aValue < bValue) {
+        return queryParams.sortOrder === SortOrder.desc ? 1 : -1;
+      }
+      if (aValue > bValue) {
+        return queryParams.sortOrder === SortOrder.asc ? -1 : 1;
+      }
+      return 0;
+    });
+  }, [tableData, queryParams]);
+
+  const paginatedSortedTableData: InferenceEndpointUI[] = useMemo(() => {
+    const startIndex = pagination.pageIndex * pagination.pageSize;
+    const endIndex = startIndex + pagination.pageSize;
+    return sortedTableData.slice(startIndex, endIndex);
+  }, [sortedTableData, pagination]);
+
+  return (
+    <InferenceEndpointsProvider>
+      <EuiPageTemplate offset={0} restrictWidth={false} grow={false}>
+        <EuiPageTemplate.Header
+          css={{ '.euiPageHeaderContent > .euiFlexGroup': { flexWrap: 'wrap' } }}
+          pageTitle={i18n.translate(
+            'xpack.searchInferenceEndpoints.inferenceEndpoints.allInferenceEndpoints.title',
+            {
+              defaultMessage: 'Inference endpoints',
+            }
+          )}
+          description={i18n.translate(
+            'xpack.searchInferenceEndpoints.inferenceEndpoints.allInferenceEndpoints.description',
+            {
+              defaultMessage:
+                'Manage your Elastic and third-party endpoints generated from the Inference API.',
+            }
+          )}
+          rightSideItems={[
+            <EuiFlexGroup gutterSize="xs">
+              <EuiFlexItem>
+                <EuiButton
+                  key="newInferenceEndpoint"
+                  color="primary"
+                  iconType="plusInCircle"
+                  fill
+                  onClick={() => setIsInferenceFlyoutVisible(true)}
+                >
+                  {addEndpointLabel}
+                </EuiButton>
+              </EuiFlexItem>
+            </EuiFlexGroup>,
+          ]}
+        />
+        <EndpointsTable
```

Comment:
This should be wrapped in a page section to have the proper layout. [example here](https://github.com/elastic/kibana/blob/main/x-pack/plugins/search_playground/public/components/app.tsx#L18) which uses the `<KibanaPageTemplate.Section>`, or you can use a `<EuiPageTemplate.Section>`